src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripScheduleWithOffset.java

```diff
@@ -38,21 +44,32 @@ public int tripSortIndex() {
 
     @Override
     public int arrival(int stopPosInPattern) {
-        return this.tripTimes.getArrivalTime(stopPosInPattern) + secondsOffset;
+        return this.pattern.getArrivalTimes(stopPosInPattern).applyAsInt(tripIndexForDates);
```

Comment:
It might make sense to create methods like `getDepartureTimesForTrip` and `getArrivalTimesForTrip` in `TripPatternForDates`, so we could create only two lambdas in the constructor and use them directly here.

Reply:
Small change, but seems indeed slightly faster
```
With suggested change:
Worker: 
 ==> mc_destination : [  533,  514,  512,  516 ] Avg: 518.8

Total:  
 ==> mc_destination : [  737,  708,  705,  709 ] Avg: 714.8
 
Before:
Worker: 
 ==> mc_destination : [  534,  524,  520,  516 ] Avg: 523.5

Total:  
 ==> mc_destination : [  744,  720,  715,  710 ] Avg: 722.3

```